vcd/resource_vcd_ip_space.go

```diff
@@ -180,9 +192,37 @@ func resourceVcdIpSpaceUpdate(ctx context.Context, d *schema.ResourceData, meta
 	ipSpaceConfig.ID = d.Id()
 	_, err = ipSpace.Update(ipSpaceConfig)
 	if err != nil {
+		// Duplicate key case can happen when splitting a single IP range
+		// start_address = "11.11.11.100"
+		// end_address   = "11.11.11.111"
+		// Into two ranges, where one matches `start_address` and another matches `end_address`
+		// start_address = "11.11.11.100" <- matches the start_address of single range I had
+		// end_address   = "11.11.11.108"
+		// start_address = "11.11.11.110"
+		// end_address   = "11.11.11.111" <- matches the end_address of single range I had
+		//
+		// It does not cause any issues on VCD or Terraform state except the split is not done even
+		// after multiple apply attempts.
+		// The point here is to hint user that what he is doing could be done in two steps
+		if strings.Contains(err.Error(), "Duplicate key") {
+			return diag.Errorf("error updating IP Space: splitting a single IP Range into two ranges "+
+				"with matching 'start_address' and 'end_address' is not supported in a single update "+
+				"operation. Please perform it in two steps: %s", err)
+		}
+
 		return diag.Errorf("error updating IP Space: %s", err)
 	}
 
+	// Operations on IP Space related entities trigger a separate task
+	// 'ipSpaceUplinkRouteAdvertisementSync' which is better to finish before any other operations
+	// as it might cause an error: busy completing an operation IP_SPACE_UPLINK_ROUTE_ADVERTISEMENT_SYNC
+	// Sleeping a few seconds because the task is not immediately seen sometimes.
+	time.Sleep(3 * time.Second)
```

Comment:
Is there any risk that 3 seconds is not enough?

Reply:
I hope not - I added this as an extra precaution. In some cases, the task even finishes within this timeframe :)

Reply:
Is there any way to avoid the sleep and wait for the needed state? :)